consensus/consensus.go

```diff
@@ -52,12 +52,13 @@ type Consensus interface {
 }
 
 type ConsensusInfo struct {
-	Validators       []types.Address
-	ValidatorKey     *ecdsa.PrivateKey
-	ValidatorAddress types.Address
-	Epoch            uint64
-	QuorumSize       int
-	SetSignedPayload SetSignedPayloadFn
+	Validators            []types.Address
+	ValidatorKey          *ecdsa.PrivateKey
+	ValidatorAddress      types.Address
+	Epoch                 uint64
+	QuorumSize            int
+	SetSignedPayload      SetSignedPayloadFn
+	CustomContractAddress string
```

Comment:
This could be a types.Address too so we don't have to convert from a string later on..